src/Microsoft.ServiceFabric.AspNetCore.Configuration/ServiceFabricConfigurationOptions.cs

```diff
@@ -0,0 +1,129 @@
+    using System.Fabric.Description;
+    using System.Runtime.InteropServices;
+    using Microsoft.Extensions.Configuration;
+    using FabricConfigurationSection = System.Fabric.Description.ConfigurationSection;
+
+    /// <summary>
+    /// The options used to configure the behavior of mapping configuration package to IConfiguration items.
+    /// </summary>
+    public class ServiceFabricConfigurationOptions
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ServiceFabricConfigurationOptions"/> class.
+        /// </summary>
+        /// <param name="packageName">the name of the configuration package.</param>
+        public ServiceFabricConfigurationOptions(string packageName)
+        {
+            this.PackageName = packageName ?? throw new ArgumentNullException(packageName);
+            this.IncludePackageName = true;
+            this.DecryptValue = false; // secure by default.
+
+            this.ConfigAction = this.DefaultConfigDelegate;
+            this.ExtractKeyFunc = this.DefaultExtractKeyFunc;
+            this.ExtractValueFunc = this.DefaultExtractValueFunc;
+        }
+
+        /// <summary>
+        /// Gets the name of the configuration package this options is used for.
+        /// </summary>
+        public string PackageName { get; }
+
+        /// <summary>
+        /// Gets or sets the configuration used to transform the configuration package to IConfiguration items.
+        /// </summary>
+        /// <remarks>
+        /// This is for advanced usage scenario to take data from package and populate into the dictionary.
+        /// the default value is a delegate which used SectionName and ParamName as the key and value as the param value regardless of encrypted or not.
+        /// </remarks>
+        public Action<ConfigurationPackage, IDictionary<string, string>> ConfigAction { get; set; }
+
+        /// <summary>
+        /// Gets or sets the function to extract the IConfiguration key from the fabric configuration section and property.
+        /// </summary>
+        public Func<FabricConfigurationSection, ConfigurationProperty,  string> ExtractKeyFunc { get; set; }
+
+        /// <summary>
+        /// Gets or sets the function to extract the IConfiguration value from the fabric configuration section and property.
+        /// </summary>
+        public Func<FabricConfigurationSection, ConfigurationProperty, string> ExtractValueFunc { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether or not to include the configuration package name to the section name of IConfiguration item.
+        /// The default is true, and will include the package name, section name and param namein configuraton package.
+        /// You could set this to true for multiple config packages to avoid a setting with same section name and param name being override from other package.
+        /// </summary>
+        public bool IncludePackageName { get; set; }
```

Comment:
>IncludePackageName [](start = 20, length = 18)

Thinking more about, WE can just remove this flag and always add config package name to the key, if a separate behavior is needed, user can always provide the ExtractKeyFunc. This would keep the default usage simple.

Reply:
thanks @RamjotSingh  and @amanbha , feedback received so I remove the condition logic on the default and always make default to true.

I'm stuggle on whether to completely remove the property of IncludePackageName, I think it still make sense to have IncludePackageName, given it is default to be true, it won't causing confusing. If we need to remove the IncludePackageName I would then also consider remove DecryptValue flag as the same reason could be given that user could provide ExtractValueFunc.

Reply:
I understand your thinking about keeping it, If you want to keep the property , its fine, but I would suggest to remove the statement in ServiceFabricConfigurationExtensions to set it to false when there is only one config package. This way user will nto have to change their code when they add more config packages.

---
In reply to: [219564022](https://github.com/Microsoft/service-fabric-aspnetcore/pull/58#discussion_r219564022) [](ancestors = 219564022)

Reply:
Sure, I already removed that in extensions and fixed the UT, will send new PR soon.